[lldb] Add missing operations to GetOpcodeDataSize

JDevlieghere opened the pull request:
The improved error reporting in #120162 revealed that we were missing
opcodes in GetOpcodeDataSize. I changed the function to remove the
default case and switch over the enum type which will cause the compiler
to emit a warning if there are unhandled operations in the future.

rdar://139705570

adrian-prantl commented on lldb/source/Expression/DWARFExpression.cpp:
```diff
@@ -269,6 +269,7 @@ static lldb::offset_t GetOpcodeDataSize(const DataExtractor &data,
 
   // All opcodes that have a single ULEB (signed or unsigned) argument
   case DW_OP_addrx:           // 0xa1 1 ULEB128 index
+  case DW_OP_constx:          // 0xa2 1 ULEB128 index
```
I think there might be value in keeping the list sorted by their numeric value.

adrian-prantl commented on lldb/source/Expression/DWARFExpression.cpp:
```diff
@@ -307,6 +308,8 @@ static lldb::offset_t GetOpcodeDataSize(const DataExtractor &data,
+  case DW_OP_reinterpret:     // 0xa9 1 ULEB128 offset
   case DW_OP_GNU_addr_index:  // 0xfb 1 ULEB128 index
   case DW_OP_GNU_const_index: // 0xfc 1 ULEB128 index
     data.Skip_LEB128(&offset);
```
Should we list all values explicitly and get rid of the default so we get a warning about missing entries in the future?

JDevlieghere committed:
[lldb] Add missing operations to GetOpcodeDataSize

The improved error reporting in #120162 revealed that we were missing
opcodes in GetOpcodeDataSize. I changed the function to remove the
default case and switch over the enum type which will cause the compiler
to emit a warning if there are unhandled operations in the future.

rdar://139705570

adrian-prantl commented on lldb/source/Expression/DWARFExpression.cpp:
```diff
@@ -327,27 +358,45 @@ static lldb::offset_t GetOpcodeDataSize(const DataExtractor &data,
+  case DW_OP_implicit_pointer: // 0xa0 4 + LEB128
+  {
+    data.Skip_LEB128(&offset);
+    return 4 + offset - data_offset;
```
> The first operand is a 4-byte unsigned value in the 32-bit DWARF format, or an 8-byte unsigned value in the 64-bit DWARF format

adrian-prantl reviewed (commented):
I found one bug, otherwise this looks good!

JDevlieghere committed:
Fix DW_OP_implicit_pointer
